src/Catalyst.Core.Lib/P2P/IO/Observers/PingResponseObserver.cs

```diff
@@ -26,30 +26,36 @@
 using System.Reactive.Subjects;
 using Catalyst.Abstractions.IO.Messaging.Correlation;
 using Catalyst.Abstractions.IO.Observers;
-using Catalyst.Abstractions.P2P;
 using Catalyst.Abstractions.P2P.IO;
 using Catalyst.Abstractions.P2P.IO.Messaging.Dto;
+using Catalyst.Abstractions.P2P.Protocols;
 using Catalyst.Core.Lib.IO.Observers;
 using Catalyst.Core.Lib.P2P.IO.Messaging.Dto;
+using Catalyst.Core.Lib.P2P.Protocols;
 using Catalyst.Protocol.IPPN;
 using Catalyst.Protocol.Peer;
 using DotNetty.Transport.Channels;
 using Serilog;
 
 namespace Catalyst.Core.Lib.P2P.IO.Observers
 {
+    /// <summary>
+    ///     @TODO we inject IPeerChallengeRequest, this class probably shouldnt care about IPeerChallengeRequest,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4600656)